src/pycram/yarp_utils/yarp_joint_state_updater.py

```diff
@@ -0,0 +1,324 @@
+import math
+import sys
+from time import sleep
+
+from pycram import World
+from pycram.robot_description import RobotDescription
+from pycram.yarp_utils.yarp_networking import interrupt_and_close
+import yarp
+
+
+class icub_state_updater(yarp.RFModule):
+
+    def __init__(self):
+        yarp.RFModule.__init__(self)
+
+        # handle port for the RFModule
+        self.handle_port = yarp.Port()
+        self.attach(self.handle_port)
+
+        # Define vars to receive an image
+        self.module_name = None
+        self.robot_name_yarp = None
+
+        self.state_torso_port_name = None
+        self.state_right_arm_port_name = None
+        self.state_left_arm_port_name = None
+        self.state_head_port_name = None
+
+        self.torso_joints_name = None
+        self.right_arm_joints_name = None
+        self.left_arm_joints_name = None
+        self.head_joints_name = None
+
+        self.state_torso_port = yarp.BufferedPortBottle()
+        self.state_right_arm_port = yarp.BufferedPortBottle()
+        self.state_left_arm_port = yarp.BufferedPortBottle()
+        self.state_head_port = yarp.BufferedPortBottle()
+
+    def configure(self, rf):
+        self.module_name = rf.check("name",
+                                    yarp.Value("pycramICubStatePublisher"),
+                                    "module name (string)").asString()
+
+        self.robot_name_yarp = rf.check("robot",
+                                    yarp.Value("icub"),
+                                    "module name (string)").asString()
+
+        self.state_torso_port_name = self.getName("/state/torso:i")
+        self.state_right_arm_port_name = self.getName("/state/right_arm:i")
+        self.state_left_arm_port_name = self.getName("/state/left_arm:i")
+        self.state_head_port_name = self.getName("/state/head:i")
+
+
+        # Create handle port to read message
+        if not self.handle_port.open('/' + self.module_name):
+            print("Can't open the port correctly")
```

Comment:
Use pycram.ros.logging instead of prints, also all prints below. I really like the precise checking if all ports can be  opened and the corresponding error logging

Reply:
done